File: derf.py
import numpy as np
DERF_SYMBOL = "߈"
DERF_DIGITS = ("0","1","2","3","4","5",DERF_SYMBOL,"6","7","8","9")

class derf_int():
    def __init__(self,b10_int):
        self.b10 = b10_int
        self.b11 = np.base_repr(b10_int,11)
        self.derf = b11_to_derf(self.b11)

    def __str__(self):
        return self.derf
    
    def __add__(self,other):
        return derf_int(self.b10 + other.b10)
    
    def __sub__(self,other):
        return derf_int(self.b10 - other.b10)
    
    def __mul__(self,other):
        return derf_int(self.b10 * other.b10)
    
    def __floordiv__(self,other):
        return derf_int(self.b10 // other.b10)

def b10_to_derf(input: int):
    return b11_to_derf(np.base_repr(input,11))

def b11_to_derf(input: str):
    output = ""
    for i in range(len(input)):
        if input[i] == "A":
            output = output + DERF_DIGITS[10]
        else: output = output + DERF_DIGITS[int(input[i])]
    return output

File: derf.js
export class DerfInt {
  constructor(b10) {
    this.b10Int = b10;
    this.derfInt = Derf.b10ToDerf(this.b10Int);
  }
}

export const Derf = {
  b10ToDerf(b10) {
    if (b10 === 0) return "0";
    let b = 11;
    let b11 = "";
    while (b10 > 0) {
      b11 = this.meta.DERF_DIGITS[b10 % b] + b11;
      b10 = Math.floor(b10 / b);
    }
    return b11;
  },
  derfToB10(derf_str) {
    let exp = 1;
    let b10 = 0;
    let dig;
    for (let i=(derf_str.length -1); i>=0; i--) {
      dig = this.meta.DERF_DIGITS.indexOf(derf_str[i]);
      b10 = b10 + (dig * exp)
      exp = exp * 11;
    }
    return b10;
  },
  derfAdd(op1, op2) {
    return new DerfInt(op1.b10Int + op2.b10Int);
  },
  derfSub(op1, op2) {
    return new DerfInt(op1.b10Int - op2.b10Int);
  },
  derfMul(op1, op2) {
    return new DerfInt(op1.b10Int * op2.b10Int);
  },
  derfDiv(op1, op2) {
    return new DerfInt(Math.floor(op1.b10Int / op2.b10Int));
  },
  derfMod(op1, op2) {
    return new DerfInt(op1.b10Int % op2.b10Int);
  },
  meta: Object.freeze({
    DERF_DIGITS: [
      "0",
      "1",
      "2",
      "3",
      "4",
      "5",
      "d",
      "6",
      "7",
      "8",
      "9"
    ]
  })
};